src/tools/perf/lib/uct_tests.cc

```diff
@@ -627,44 +615,60 @@ class uct_perf_test_runner {
         }
     }
 
+ static ucs_status_t perf_memset(void *buffer, uint8_t value, size_t length,
+                                    ucs_memory_type memory_type, uct_ep_h ep,
+                                    uct_rkey_t rkey)
+    {
+        const std::vector<uint8_t> values(length, value);
+        if(memory_type == UCS_MEMORY_TYPE_HOST)
+        {
+            memset(buffer, value, length);
+            return UCS_OK;
+        }
+
+        return uct_ep_put_short(ep, values.data(), length, (uint64_t)buffer, rkey);
+    }
+
     ucs_status_t run_stream_req_uni(bool flow_control, bool send_window,
                                     bool direction_to_responder)
     {
-        const ucx_perf_allocator_t *recv_allocator;
         ucs_memory_type_t recv_mem_type;
         volatile psn_t *recv_sn;
         unsigned my_index;
         unsigned length;
         unsigned group_size;
         unsigned peer_index;
+        uct_rkey_t rkey;
+        uct_ep_h ep;
 
         group_size = rte_call(&m_perf, group_size);
 
         length = ucx_perf_get_message_size(&m_perf.params);
         ucs_assert(length >= sizeof(psn_t));
         ucs_assert(m_perf.params.uct.fc_window <= ((psn_t)-1) / 2);
 
-        m_perf.send_allocator->memset(m_perf.send_buffer, 0, length);
-        m_perf.recv_allocator->memset(m_perf.recv_buffer, 0, length);
+        my_index   = rte_call(&m_perf, group_index);
+        peer_index = rte_peer_index(group_size, my_index);
+        ep         = m_perf.uct.peers[peer_index].ep;
+        rkey       = m_perf.uct.peers[peer_index].rkey.rkey;
+        perf_memset(m_perf.send_buffer, 0, length, m_perf.uct.send_mem.mem_type,
```

Comment:
maybe don't pass value (0), rename perf_memset to init_buffer and it will always set to 0